iscript/src/iscript/script.py

```diff
@@ -8,52 +8,79 @@
+    return behavior
+
+
+def get_behavior_function(behavior):
+    """Map a behavior to a function.
+
+    Args:
+        behavior (str): signing behavior
+
+    Returns:
+        tuple: (func, {args})
+    """
+    functions = {
+        "mac_geckodriver": (single_file_behavior, {}),
+        "mac_single_file": (single_file_behavior, {}),
+        "mac_notarize": (notarize_behavior, {}),
+        "mac_notarize_vpn": (vpn_behavior, {"notarize": True}),
+        "mac_sign_and_pkg_vpn": (vpn_behavior, {"notarize": False}),
+        "mac_notarize_part_1": (notarize_1_behavior, {}),
+        "mac_notarize_part_3": (notarize_3_behavior, {}),
+        "mac_sign": (sign_behavior, {}),
+        # For staging releases; or should we mac_notarize but skip notarization
+        # for dep?
+        "mac_sign_and_pkg": (sign_and_pkg_behavior, {}),
+    }
+    if behavior not in functions:
+        raise IScriptError("iscript behavior {} not implemented!".format(behavior))
+    return functions[behavior]
+
+
 async def async_main(config, task):
     """Sign all the things.
 
     Args:
         config (dict): the running config.
         task (dict): the running task.
-
     """
     await run_command(["hostname"])
     base_key = "mac_config"  # We may support ios_config someday
     sign_config = get_sign_config(config, task, base_key=base_key)
     behavior = task["payload"].get("behavior", "mac_sign")
-    if behavior == "mac_notarize" and "mac_notarize" not in sign_config["supported_behaviors"] and "mac_sign_and_pkg" in sign_config["supported_behaviors"]:
-        behavior = "mac_sign_and_pkg"
-    if behavior not in sign_config["supported_behaviors"]:
-        raise IScriptError("Unsupported behavior {} given scopes {}!".format(behavior, task["scopes"]))
-    if behavior in ("mac_geckodriver", "mac_single_file"):
-        await single_file_behavior(config, task)
-        return
-    elif behavior == "mac_notarize":
-        await notarize_behavior(config, task)
-        return
-    elif behavior == "mac_notarize_vpn":
-        await notarize_vpn_behavior(config, task)
-        return
-    elif behavior == "mac_notarize_part_1":
-        await notarize_1_behavior(config, task)
-        return
-    elif behavior == "mac_notarize_part_3":
-        await notarize_3_behavior(config, task)
-        return
-    elif behavior == "mac_sign":
-        await sign_behavior(config, task)
-        return
-    elif behavior == "mac_sign_and_pkg":
-        # For staging releases; or should we mac_notarize but skip notarization
-        # for dep?
-        await sign_and_pkg_behavior(config, task)
-        return
-    raise IScriptError("Unknown iscript behavior {}!".format(behavior))
+
+    # Check for dep behaviors (skips notarization, only signs with dep keys)
+    # Raises if behavior not supported
+    behavior = check_dep_behavior(task, behavior, sign_config["supported_behaviors"])
```

Comment:
It looks like we call this and fall back to the dep behavior whether we're dep signing or not. Was that intentional? We should only fall back if the original behavior isn't in the supported behaviors, or if we're dep.

Per [this block](https://github.com/mozilla-platform-ops/ronin_puppet/blob/master/data/common.yaml#L89-L99), this would have turned off notarization for all of our prod pools.

Reply:
You'd think our tests would catch that. Nice find :)

Reply:
(I probably broke the tests btw, so still on me hahahah)